src/coordinator.rs

```diff
@@ -17,9 +17,9 @@ pub struct Coordinator {
 }
 
 impl Coordinator {
-    pub fn new(workspace_root: PathBuf, store: Store) -> Self {
+    pub fn new(workspace_root: &Path, store: Store) -> Self {
```

Comment:
nit general design comment: if you are just gonna take ownership or call `to_path_buf`/`to_owned`/`to_string`/etc, you should take the final type you want as an argument to the function. So in this case, directly take the `PathBuf`. That way, if the caller can give you ownership, no wasted copy is needed. If the caller can't, they can explicitly request the copy. Helps reduce wasted work from accidentally copies if you ultimately want ownership of a value.

Reply:
that's super helpful design advice, thank you!